python/GafferUI/ColorChooserPlugValueWidget.py

```diff
@@ -0,0 +1,120 @@
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import imath
+
+import Gaffer
+import GafferUI
+
+class ColorChooserPlugValueWidget( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, plugs, **kw ) :
+
+		self.__colorChooser = GafferUI.ColorChooser()
+
+		GafferUI.PlugValueWidget.__init__( self, self.__colorChooser, plugs, **kw )
+
+		self.__colorChangedConnection = self.__colorChooser.colorChangedSignal().connect(
+			Gaffer.WeakMethod( self.__colorChanged ), scoped = False
+		)
+
+		self.visibilityChangedSignal().connect( Gaffer.WeakMethod( self.__visibilityChanged ), scoped = False )
+
+		self.__lastChangedReason = None
+		self.__mergeGroupId = 0
+
+		self._updateFromPlugs()
+
+	def setPlugs( self, plugs ) :
+
+		GafferUI.PlugValueWidget.setPlugs( self, plugs )
+
+		self.__colorChooser.setInitialColor( self.__colorFromPlugs() )
+
+	def _updateFromPlugs( self ) :
+
+		with Gaffer.BlockedConnection( self.__colorChangedConnection ) :
+			self.__colorChooser.setColor( self.__colorFromPlugs() )
+
+		self.__colorChooser.setEnabled( self.__allComponentsEditable() )
+
+	def __colorChanged( self, colorChooser, reason ) :
+
+		if not GafferUI.ColorChooser.changesShouldBeMerged( self.__lastChangedReason, reason ) :
+			self.__mergeGroupId += 1
+		self.__lastChangedReason = reason
+
+		with Gaffer.UndoScope(
+			next( iter( self.getPlugs() ) ).ancestor( Gaffer.ScriptNode ),
+			mergeGroup = "ColorPlugValueWidget%d%d" % ( id( self, ), self.__mergeGroupId )
+		) :
+
+			with Gaffer.BlockedConnection( self._plugConnections() ) :
+				for plug in self.getPlugs() :
+					plug.setValue( self.__colorChooser.getColor() )
+
+	def __colorFromPlugs( self ) :
+
+		with self.getContext() :
+			values = [ p.getValue() for p in self.getPlugs() ]
+
+		if len( values ) :
+			# ColorChooser only supports one colour, and doesn't have
+			# an "indeterminate" state, so when we have multiple plugs
+			# the best we can do is take an average.
+			return sum( values ) / len( values )
```

Comment:
Hmm, I'm trying to think of where this averaging would actually be triggered, and wondering if some sort of error state might be better? If you had multiple plugs that were similar but not identical in color, seeing an average might confuse you into thinking they are the same?

Reply:
It's triggered when editing multiple spreadsheet cells at once, or when editing multiple lights at once in the SceneViewInspector. In both those cases we have been performing the same averaging from the outset - see ColorSwatchPlugValueWidget. This just applies that approach to the new ColorChooserPlugValueWidget.